src/main/java/com/google/devtools/build/lib/vfs/inmemoryfs/InMemoryFileInfo.java

```diff
@@ -57,14 +58,19 @@ public byte[] getxattr(String name) {
 
   @Override
   public byte[] getFastDigest() {
-    return null;
+    return fastDigest;
   }
 
   private synchronized void setContent(byte[] newContent) {
     content = newContent;
+    fastDigest = null; // assume that content has changed and the existing fastDigest is invalid
     markModificationTime();
   }
 
+  public synchronized void setFastDigest(byte[] newDigest) {
```

Comment:
IIUC, this method and field `fastDigest` are added only because of test. Please instead do the mock in the test. Adding a field here will probably increase the memory footprint a lot since the inmemory file node are created intensively in some cases.